src/PusherCrypto.php

```diff
@@ -109,6 +128,111 @@ public function decrypt_event(object $event): object
         return $event;
     }
 
+    /**
+     * Encode multiple channel names into a parseable header
+     *
+     * @param array $channels
+     * @param string $random
+     * @return string
+     *
+     * @throws \SodiumException
+     */
+    public function multi_channel_encode(array $channels, string $random = ''): string
+    {
+        // Determine a stable order of channel names:
+        $sorted = array_values($channels);
+        sort($sorted);
+
+        $list = [];
+        $pos = strlen(self::ENCRYPTED_PREFIX);
+        foreach ($sorted as $ch) {
+            if (!self::is_encrypted_channel($ch)) {
+                continue;
```

Comment:
We may want to `throw` here instead. The intent for this method is to _only_ allow encrypted channels.